File: Backend/api/v1/tools/health.py
from fastapi import APIRouter
from datetime import datetime
import sys

from schemas import HealthCheckResponse, HealthInfoResponse
from core.config import settings

router = APIRouter(prefix="/health", tags=["health"])


@router.get("", response_model=HealthCheckResponse)
async def health_check():
    """基础健康检查"""
    return HealthCheckResponse(
        status="healthy",
        version=settings.app_version,
        timestamp=datetime.now().isoformat()
    )


@router.get("/info", response_model=HealthInfoResponse)
async def health_info():
    """详细健康信息"""
    return HealthInfoResponse(
        status="healthy",
        version=settings.app_version,
        name=settings.app_name,
        description=settings.app_description,
        python_version=sys.version,
        timestamp=datetime.now().isoformat()
    )

File: Backup/tools_project_rust/src/routes/health.rs
use axum::{routing::get, Extension, Json, Router};
use chrono::Local;
use serde_json::json;

use crate::core::config::SETTINGS;
use crate::core::db;
use crate::models::{BaseInfo, BaseResponse, HealthCheckResponse, HealthInfoResponse};

pub fn router() -> Router {
    Router::new()
        .route("/", get(health_check))
        .route("/info", get(health_info))
        .route("/db", get(db_health_check))
}

async fn health_check(
    Extension(base): Extension<Option<BaseInfo>>,
) -> Json<BaseResponse<HealthCheckResponse>> {
    Json(BaseResponse::ok(
        HealthCheckResponse {
            status: "healthy".to_string(),
            version: SETTINGS.app_version.clone(),
            timestamp: Local::now().to_rfc3339(),
        },
        base,
    ))
}

async fn health_info(
    Extension(base): Extension<Option<BaseInfo>>,
) -> Json<BaseResponse<HealthInfoResponse>> {
    Json(BaseResponse::ok(
        HealthInfoResponse {
            status: "healthy".to_string(),
            version: SETTINGS.app_version.clone(),
            name: SETTINGS.app_name.clone(),
            description: SETTINGS.app_description.clone(),
            rust_version: format!("rustc {}", option_env!("CARGO_PKG_RUST_VERSION").unwrap_or("unknown")),
            timestamp: Local::now().to_rfc3339(),
        },
        base,
    ))
}

async fn db_health_check(
    Extension(base): Extension<Option<BaseInfo>>,
) -> Json<BaseResponse<serde_json::Value>> {
    match db::get_pool() {
        Some(pool) => match db::ping(pool).await {
            Ok(_) => Json(BaseResponse::ok_with_message(
                json!({ "status": "connected" }),
                "数据库连接正常",
                base,
            )),
            Err(e) => Json(BaseResponse::err(
                format!("数据库连接异常: {}", e),
                base,
            )),
        },
        None => Json(BaseResponse::err(
            "数据库连接池未初始化",
            base,
        )),
    }
}
